shell/core/types.ts

```diff
@@ -610,6 +611,13 @@ export interface IPlugin {
    */
   setHomePage(component: any): void;
 
+  /**
+   * Set 'onReady' function that will be called once the user has logged in and the UI is ready
+   *
+   * @param onReady Function to be called when the user has logged in and the UI is ready
+   */
+  setOnReady(onReady: OnReady): void;
```

Comment:
we have this below

```
 addNavHooks(
    onEnter?: OnNavToPackage,
    onLeave?: OnNavAwayFromPackage,
    onLogOut?: OnLogOut
  ): void;
```

given this is on logged in and ready shouldn't this be an `onReady` in there instead?